components/org.wso2.carbon.identity.provisioning.connector.google/src/test/java/org/wso2/carbon/identity/provisioning/connector/google/GoogleProvisioningConnectorTest.java

```diff
@@ -0,0 +1,521 @@
+        Property[] provisioningProperties = new Property[]{property1, property2, property3};
+
+        GoogleProvisioningConnector connector = getConnector(provisioningProperties);
+        Path path = Paths.get("src", "test", "resources", "googlePrvKey");
+        Whitebox.setInternalState(GoogleProvisioningConnector.class, "googlePrvKey", path.toFile());
+        mockStatic(SecurityUtils.class);
+        PrivateKey privateKey = mock(PrivateKey.class);
+        when(SecurityUtils.loadPrivateKeyFromKeyStore(any(KeyStore.class), any(InputStream.class), anyString(),
+                                                         anyString(), anyString())).thenReturn(privateKey);
+
+        Directory directoryService = connector.getDirectoryService();
+        Assert.assertNotNull(directoryService, "Directory Service cannot be null.");
+    }
+
+    @DataProvider(name = "updateGoogleUserDataProvider")
+    public Object[][] provideUpdateGoogleUserTestData() {
+
+        String claimUri = "org:wso2:carbon:identity:provisioning:claim:custom claim";
+        ClaimMapping claimMapping = new ClaimMapping();
+        Claim claim = new Claim();
+        claim.setClaimUri(claimUri);
+        claimMapping.setRemoteClaim(claim);
+
+        Map<ClaimMapping, List<String>> attributes1 = new HashMap<>();
+        attributes1.put(claimMapping, Collections.singletonList(username));
+        Map<ClaimMapping, List<String>> attributes2 = new HashMap<>();
+
+        return new Object[][]{
+                {true, attributes1},
+                {false, attributes1},
+                {false, attributes2}
+        };
+    }
+
+    @Test(dataProvider = "updateGoogleUserDataProvider")
+    public void testUpdateGoogleUser(boolean debugEnabled, Object attributesObj) throws Exception {
+
+        String defaultFamilyNameKey = "google_prov_familyname";
+        String defaultGivenNameKey = "google_prov_givenname";
+        String defaultFamilyName = "defaultFamilyName";
+        String defaultGivenName = "defaultGivenName";
+        String email = "testuser@gmail.com";
+        Map<ClaimMapping, List<String>> attributes = (Map<ClaimMapping, List<String>>) attributesObj;
+        setLogging(debugEnabled);
+
+        ProvisioningEntity entity = new ProvisioningEntity(ProvisioningEntityType.USER, ProvisioningOperation.POST,
+                                                           attributes);
+        ProvisionedIdentifier identifier = new ProvisionedIdentifier();
+        identifier.setIdentifier(email);
+        entity.setIdentifier(identifier);
+
+        Property property1 = buildProperty(defaultFamilyNameKey, defaultFamilyName);
+        Property property2 = buildProperty(defaultGivenNameKey, defaultGivenName);
+        Property[] provisioningProperties = new Property[]{property1, property2};
+
+        GoogleProvisioningConnector connector = getConnector(provisioningProperties);
+        User user = connector.updateGoogleUser(entity);
+
+        if (!attributes.isEmpty()) {
```

Comment:
Would be better to check whether we can avoid this branching by sending the expected result from the data provider

Reply:
One case includes `user` object being null. So better to stick to this format